src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -216,4 +216,39 @@ public void quarantine(
       adaptor.delete(datasetId);
     }
   }
+
+  @Override
+  public void unstage(
+      @Pattern(regexp = "^[0-9a-zA-Z]+$") @Size(min = 36, max = 36) final String datasetId)
+      throws DatasetNotFoundException, QuarantineException {
+    fileStorageAdaptor.updateStatus(datasetId, STORED);
+    metacardStorageAdaptor.delete(datasetId);
+    log.info("Unstaged file and deleted staged metacard for datasetId={{}}", datasetId);
+  }
+
+  private void verifyStoreStatus(String datasetId, StorageAdaptor adaptor) {
+    String status = adaptor.getStatus(datasetId);
+    // TODO: Should the exception that is returned depend on what the status is (Staged vs
+    // Quarantined)?
```

Comment:
❓ Should you use !StringUtils.equalsIgnoreCase(status, STORED) instead or an enum just to be safe?

Reply:
Most people have suggested an enum, so I'll go ahead and change that.

Reply:
I'm going through and updating to an enum, but I just realized that doesn't fix the check here. The storage adaptor shouldn't take or return an enum. We're stuck with string comparisons for this one.